napari/benchmarks/benchmark_shapes_layer.py

```diff
@@ -212,6 +219,256 @@ def time_select_shape(self, n):
     time_select_shape.param_names = ['n_shapes']
 
 
+def get_shape_type(func):
+    @wraps(func)
+    def wrap(self, *args):
+        shape_type_pos = self.param_names.index('shape_type')
+        return func(self, args[shape_type_pos])
+
+    return wrap
+
+
+class _ShapeTriangulationBase:
+    def select_backend(self, compiled_triangulation):
+        with suppress(AttributeError):
+            self.prev_settings = (
+                get_settings().experimental.compiled_triangulation
+            )
+            get_settings().experimental.compiled_triangulation = (
+                compiled_triangulation
+            )
+
+        from napari.layers.shapes import _shapes_utils
+
+        self.triangulate = _shapes_utils.triangulate
+        _shapes_utils.triangulate = None
+
+    def revert_backend(self):
+        with suppress(AttributeError):
+            get_settings().experimental.compiled_triangulation = (
+                self.prev_settings
+            )
+
+        from napari.layers.shapes import _shapes_utils
+
+        _shapes_utils.triangulate = self.triangulate
+
+    def teardown(self, *_):
+        self.revert_backend()
+
+    @get_shape_type
+    def time_create_layer(self, shape_type):
+        """Time to create a layer."""
+        Shapes(self.data, shape_type=shape_type)
+
+    @get_shape_type
+    def time_create_shapes(self, shape_type):
+        """Time to create a layer."""
+        cls = shape_classes[shape_type]
+        for data in self.data:
+            cls(data)
+
+
+class _ShapeTriangulationBaseShapeCount(_ShapeTriangulationBase):
+    param_names = [
+        'n_shapes',
+        'n_points',
+        'shape_type',
+        'compiled_triangulation',
+    ]
+    params = [
+        (
+            100,
+            5_000,
+        ),
+        (8, 32, 128),
+        ('path', 'polygon'),
+        (True, False),
+    ]
+
+
+class ShapeTriangulationNonConvexSuite(_ShapeTriangulationBaseShapeCount):
+    skip_params = Skip(
+        always=lambda n_shapes,
+        n_points,
+        shape_type,
+        compiled_triangulation: n_shapes == 5000
+        and n_points == 128
+        and shape_type == 'polygon'
+        and not compiled_triangulation
```

Comment:
Skip because of the bug reported here https://github.com/vispy/vispy/issues/2648